tests/post/steps/test_service_configuration.py

```diff
@@ -0,0 +1,271 @@
+                )
+            )
+
+        assert int(value) == eval("response"+"{}".format(path)), (
+            "Expected value {} for service {}, got {}".format(
+                value, service, eval("response"+"{}".format(path))
+            )
+        )
+    utils.retry(
+        _wait_for_deployment,
+        times=10,
+        wait=5,
+        name="wait for available Deployments"
+    )
+
+
+# }}}
+
+
+# Helpers {{{
+
+
+def read_namespace_config_map(k8s_client, name, namespace):
+    try:
+        response = k8s_client.read_namespaced_config_map(
+            name, namespace
+        )
+    except Exception as exc:
+        pytest.fail(
+            "Unable to read {} ConfigMap  in namespace {} with error: {!s}"
+            .format(name, namespace, exc)
+        )
+
+    return response
+
+
+def patch_namespace_config_map(k8s_client, name, namespace, patch):
+    try:
+        response = k8s_client.patch_namespaced_config_map(
+            name, namespace, patch
+        )
+    except Exception as exc:
+        pytest.fail(
+            "Unable to patch ConfigMap {} in namespace {} with error {!s}"
+            .format(name, namespace, exc)
+        )
+
+    return response
+
+
+def yaml_load(full_csc, name, namespace):
+    try:
+        csc_obj = yaml.safe_load(full_csc.data['config.yaml'])
+    except yaml.YAMLError as exc:
+        raise Exception(
+            'Invalid YAML format in ConfigMap {} found in namespace {}: {!s}'
+            .format(name, namespace, exc)
+        )
+    except Exception as exc:
+        raise Exception(
+            "Failed loading `config.yaml` from ConfigMap {} in namespace {}: "
+            "{!s}".format(name, namespace, exc)
+        )
+    return csc_obj
+
+
+def traversed_dict(data, key):
+
+    '''
+    Traverse a dict using a 'delimiter' on a  target string.
+    The target 'foo:bar:baz' will return data['foo']['bar']['baz'] if this
+    value exists, and will otherwise return the dict in the default argument.
+    Same as the utility `salt.utils.data.traverse_dict`
+    '''
+    ptr = data
+    try:
+        for each in key.split('.'):
+            ptr = ptr[each]
+    except Exception as exc:
+        raise Exception(
+            "Failed generating a traversed dictionary with error: {!s}".format(
+                exc
+            )
+        )
+    return ptr
+
+
+def teardown_configmap(k8s_client):
+    name = "metalk8s-dex-config"
+    namespace = "metalk8s-auth"
+
+    # Teardown and leave service configuration ConfigMap in default state
+    csc_to_teardown = read_namespace_config_map(k8s_client, name, namespace)
+    csc_obj = yaml_load(csc_to_teardown, name, namespace)
+    csc_obj['spec']['deployment']['replicas'] = 2
+    patch = \
+        {'data': {'config.yaml': yaml.safe_dump(
+            csc_obj, default_flow_style=False)
+        }}
+    response = patch_namespace_config_map(
+        k8s_client, name, namespace, patch
+    )
+
+    return response
```

Comment:
Hmm why ? Do we really care about it ? (Also here `metalk8s-dex-config` is hardcoded not really like it)